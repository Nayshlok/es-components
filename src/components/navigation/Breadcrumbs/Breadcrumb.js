import React from 'react';
import PropTypes from 'prop-types';
import BreadcrumbAction from './BreadcrumbAction';
import Icon from '../../base/icons/Icon';

function Breadcrumb({ breadcrumbClasses, children, ...props }) {
  return (
    <div className={breadcrumbClasses}>
      {React.Children.map(children, (child, i) => (
        <span key={name}>
          {i > 0 && <Icon name="chevron-right" />}
          {child}
        </span>
      ))}
    </div>
  );
}

Breadcrumb.propTypes = {
  breadcrumbClasses: PropTypes.string,
  children: PropTypes.node.isRequired
};

Breadcrumb.BreadcrumbAction = BreadcrumbAction;

export default Breadcrumb;
