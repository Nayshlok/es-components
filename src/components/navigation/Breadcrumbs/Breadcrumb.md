Use breadcrumbs to chain a list of items together
```

<div>
    <Breadcrumb breadcrumbClasses="test">
        <Breadcrumb.BreadcrumbAction displayName="Home" Action={() => alert("hi")}/>
        <Breadcrumb.BreadcrumbAction displayName="Testing" Action={() => alert("Goodbye")}/>
        <Breadcrumb.BreadcrumbAction displayName="More" Action={() => alert("Testing McTesterson")}/>
    </Breadcrumb>

</div>

```