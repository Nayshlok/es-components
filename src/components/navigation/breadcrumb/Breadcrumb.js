import React from 'react';
import PropTypes from 'prop-types';
import Icon from '../../base/icons/Icon';

function Breadcrumb({ children, ...props }) {
  return (
    <div {...props}>
      {React.Children.map(children, (child, i) => (
        <span key={child.props.name}>
          {i > 0 && <Icon name="chevron-right" />}
          {child}
        </span>
      ))}
    </div>
  );
}

Breadcrumb.propTypes = {
  children: (props, propName, component) => {
    if (
      PropTypes.oneOfType([
        PropTypes.element,
        PropTypes.arrayOf(PropTypes.element)
      ])
    ) {
      return props[propName].every(child => child.props.name)
        ? null
        : new Error('The children must have a name attribute');
    }
    return new Error(
      'Please use an element as a child, and ensure the child has a name property'
    );
  }
};

export default Breadcrumb;
