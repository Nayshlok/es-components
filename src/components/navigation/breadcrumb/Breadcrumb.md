Use breadcrumbs to chain a list of items together
```
function Link({destination, name}){
  return <a href={`#${destination}`} style={{textDecorationLine: 'none'}}>{name}</a>
}

function Location({name}){
  return <span>{name}</span>
}

   <Breadcrumb className="test">
    <Link name="home" destination="bye there"/>
    <Button handleOnClick={() => alert("on my way")} styleType="accent">getting there</Button>
    <Link name="test" destination="bye"/>
    <Location name="destination"/>
  </Breadcrumb>

```