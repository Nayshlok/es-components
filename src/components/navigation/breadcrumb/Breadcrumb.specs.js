/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Breadcrumb from './Breadcrumb';

describe('BreadcrumbTestSuite', () => {
  let instanceToRender;

  beforeEach(() => {
    instanceToRender = (
      <Breadcrumb keySelector={child => child.props.name}>
        <a>test</a>
        <a>test2</a>
      </Breadcrumb>
    );
  });

  it('renders as expected', () => {
    const tree = renderer.create(instanceToRender).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Does not have arrow on end', () => {
    const crumbInstance = shallow(instanceToRender);
    const iconInstances = crumbInstance.find({ name: 'chevron-right' });
    expect(iconInstances).toHaveLength(1);
  });
});
