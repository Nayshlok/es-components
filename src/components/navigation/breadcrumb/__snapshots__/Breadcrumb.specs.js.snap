// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testSuite renders as expected 1`] = `
<div>
  <span>
    <a>
      test
    </a>
  </span>
  <span>
    <i
      aria-hidden={true}
      className="sc-bdVaJa izWgQO"
      content="620"
      fontFamily="oecom-icons"
      fontSize="inherit"
    />
    <a>
      test2
    </a>
  </span>
</div>
`;
